imblearn/utils/tests/test_validation.py

```diff
@@ -46,17 +46,16 @@ def test_check_ratio_error():
 
 
 def test_ratio_all_over_sampling():
+    _expected_ratio = {1: 50, 2: 0, 3: 75}
```

Comment:
I would keep it, for readability. Otherwise I would do something like:
```py
def test_ratio_all_over_sampling():
    y = np.array([1] * 50 + [2] * 100 + [3] * 25)
    for each in ('all', 'auto'):
        out =  check_ratio(each, y, 'over-sampling') 
        assert out == {1: 50, 2: 0, 3: 75}
```